tests/dualtor/test_standby_tor_downstream.py

```diff
@@ -0,0 +1,192 @@
+    standby_tor.shell('ip route replace {}/32 {}'.format(active_tor_loopback_ip, nexthop_str))
+
+
+def remove_loopback_routes(standby_tor, active_tor_loopback_ip):
+    """
+    Remove static routes for active tor's loopback
+    """
+    logger.info("Removing dual ToR peer switch loopback route")
+    standby_tor.shell('ip route del {}/32'.format(active_tor_loopback_ip), module_ignore_errors=True)
+
+
+def test_standby_tor_downstream(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo):
+    """
+    Verify tunnel traffic to active ToR is distributed equally across nexthops, and
+    no traffic is forwarded to server from standby ToR
+    """
+    params = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+    check_tunnel_balance(**params)
+
+
+def test_standby_tor_downstream_t1_link_recovered(ptfhost, rand_selected_dut, rand_unselected_dut, verify_crm_nexthop_counter_not_increased, tbinfo):
+    """
+    Verify traffic is distributed evenly after t1 link is recovered; 
+    Verify CRM that no new nexthop created
+    """
+    PAUSE_TIME = 30
+
+    down_link = shutdown_random_one_t1_link(rand_selected_dut)
+    time.sleep(PAUSE_TIME)
+    params = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+    try:
+        check_tunnel_balance(**params)
+    except Exception as e:
+        no_shutdown_t1_link(rand_selected_dut, down_link)
+        raise e
+
+    no_shutdown_t1_link(rand_selected_dut, down_link)
+    time.sleep(PAUSE_TIME)
+    params = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+    # For mocked dualtor, we should update static route manually after link recovered
+    if 't0' in tbinfo['topo']['name']:
+        remove_loopback_routes(rand_selected_dut, params['active_tor_ip'])
+        add_loopback_routes(rand_selected_dut, params['active_tor_ip'])
+    check_tunnel_balance(**params)
+
+
+def test_standby_tor_downstream_bgp_recovered(ptfhost, rand_selected_dut, rand_unselected_dut, verify_crm_nexthop_counter_not_increased, tbinfo):
+    """
+    Verify traffic is shifted to the active links and no traffic drop observed;
+    Verify traffic is distributed evenly after BGP session is recovered; 
+    Verify CRM that no new nexthop created
+    """
+    PAUSE_TIME = 30
+    
+    down_bgp = shutdown_random_one_bgp_session(rand_selected_dut)
+    time.sleep(PAUSE_TIME)
+    params = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+    try:
+        check_tunnel_balance(**params)
+    except Exception as e:
+        startup_bgp_session(rand_selected_dut, down_bgp)
+        raise e
+
+    startup_bgp_session(rand_selected_dut, down_bgp)
+    time.sleep(PAUSE_TIME)
+    params = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+    check_tunnel_balance(**params)
+
+    
+def test_standby_tor_downstream_loopback_route_removed(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo):
```

Comment:
Please combine the loopback test cases into one case (similar to the other cases above)

Reply:
Updated. Thanks